table/multiget_context.h

```diff
@@ -56,6 +56,59 @@ struct KeyContext {
   KeyContext() = default;
 };
 
+// A read Blob request structure for use in BlobSource::MultiGetBlob and
+// BlobFileReader::MultiGetBlob.
+struct BlobReadRequest {
+  // User key to lookup the paired blob
+  const Slice* user_key;
+
+  // File offset in bytes
+  uint64_t offset;
+
+  // Length to read in bytes
+  size_t len;
+
+  // Output parameter set by MultiGetBlob() to point to the data buffer, and
+  // the number of valid bytes
+  PinnableSlice* result;
+
+  // Status of read
+  Status* status;
+
+  // Blob compression type
+  CompressionType compression;
+
+  BlobReadRequest(const Slice& _user_key, uint64_t _offset, size_t _len,
+                  PinnableSlice* _result, Status* _status,
+                  CompressionType _compression = kNoCompression)
+      : user_key(&_user_key),
+        offset(_offset),
+        len(_len),
+        result(_result),
+        status(_status),
+        compression(_compression) {}
+
+  BlobReadRequest() = default;
+
+  BlobReadRequest(const BlobReadRequest& other)
```

Comment:
Hmm... just use this to setup test cases.

Reply:
These copy ctor and copy assignment operator are identical to what the compiler would generate, right? If so, we don't have to implement them ourselves.

Reply:
Another benefit of not explicitly implmenting the copy/assignment is that, once new fields are added, we do not have to update the copy/assignments.

Reply:
Done